testing/sdk_auth.py

```diff
@@ -0,0 +1,376 @@
+                        tasks=raw_tasks))
+
+
+@retry(stop_max_attempt_number=2, wait_fixed=1000)
+def _get_host_name(host_id: str) -> str:
+    """
+    Fetches the host name for the host running the KDC app.
+    :param host_id (str): The ID of the host, used to look up the appropriate node.
+    :return (str): Name of the host running the KDC app.
+    """
+    log.info("Getting host name")
+    raw_nodes = sdk_cmd.run_cli("node --json")
+    if raw_nodes:
+        nodes = json.loads(raw_nodes)
+        for node in nodes:
+            if node["id"] == host_id:
+                log.info("Host name is {host_name}".format(host_name=node["hostname"]))
+                return node["hostname"]
+
+    raise RuntimeError("Failed to get name of host running the KDC app: {nodes}")
+
+
+@retry(stop_max_attempt_number=2, wait_fixed=1000)
+def _get_master_public_ip() -> str:
+    """
+    :return (str): The public IP of the master node in the DC/OS cluster.
+    """
+    dcos_url, headers = sdk_security.get_dcos_credentials()
+    cluster_metadata_url = "{cluster_url}/metadata".format(cluster_url=dcos_url)
+    raw_response = sdk_cmd.request("GET", cluster_metadata_url, verify=False)
+    if not raw_response.ok:
+        raise RuntimeError("Unable to get the master node's public IP address: {err}".format(err=repr(raw_response)))
+
+    response = json.loads(raw_response.text)
+    if "PUBLIC_IPV4" not in response:
+        raise KeyError("Cluster metadata does not include master's public ip: {response}".format(
+            response=repr(raw_response)))
+
+    public_ip = response["PUBLIC_IPV4"]
+    log.info("Master public ip is {public_ip}".format(public_ip=public_ip))
+    return public_ip
+
+
+def _create_temp_working_dir() -> str:
+    """
+    Creates a temporary working directory to enable setup of the Kerberos environment.
+    :return (str): The path of the working directory.
+    """
+    dir_name = "/tmp/{kerberos}_{current_time}".format(kerberos=KERBEROS_APP_ID, current_time=str(time.time()))
+    log.info("Creating temp working directory {}".format(dir_name))
+    try:
+        os.makedirs(dir_name)
+    except OSError as e:
+        raise OSError("Can't create temp working directory: {}".format(repr(e)))
+
+    return dir_name
+
+
+def _copy_file_to_localhost(self):
+    """
+    Copies the keytab that was generated inside the container running the KDC server to the localhost
+    so it can be uploaded to the secret store later. This must be done in multiple steps as we have to jump through
+    hoops.
+
+    The keytab will end up in path: <temp_working_dir>/<keytab_file>
+    """
+    log.info("Copying {} to the temp working directory".format(self.keytab_file_name))
+
+    # 1. copy from within container to private agent
+    copy_cmd = "sudo cp /var/lib/mesos/slave/slaves/{host_id}/frameworks/{framework_id}/executors/{task_id}/runs/latest/{keytab_file} /home/{linux_user}/{keytab_file}".format(
```

Comment:
Could we split out the source and destination paths here giving them descriptive names?

Reply:
Done.